crates/test-programs/wasi-http-tests/src/bin/outbound_request.rs

```diff
@@ -89,44 +89,63 @@ fn request(
     })
 }
 
+fn findHeader(r: &Response, key: String) -> Option<String> {
+    for item in r.headers.iter() {
+        if item.0 == key {
+            return Some(item.1.clone());
+        }
+    }
+    None
+}
```

Comment:
since Response is defined above, I would move this function up there. A more idiomatic way to write it is, right under the definition of `struct Response`, `impl Response { fn header(&self, name: &str) -> Option<&String> { self.headers.iter().find_map(|(k, v)| if k == name { Some(v) } else { None } ) }`